clio-server/src/main/scala/org/broadinstitute/clio/server/dataaccess/LocalFilePersistenceDAO.scala

```diff
@@ -0,0 +1,16 @@
+package org.broadinstitute.clio.server.dataaccess
+
+import org.broadinstitute.clio.server.ClioServerConfig.Persistence
+
+import java.nio.file.{Files, Path}
+
+/**
+  * Persistence DAO which writes to local disk, for running Clio locally
+  * without thrashing GCS in the dev environment.
+  */
+class LocalFilePersistenceDAO(config: Persistence.LocalConfig)
+    extends PersistenceDAO {
+  override lazy val rootPath: Path = config.rootDir.getOrElse {
+    Files.createTempDirectory("clio-persistence")
```

Comment:
`LocalConfig` is also for manual testing / exploration, to avoid polluting dev storage when running a clio-server locally and trying things out. I made the local root directory optional to avoid breaking local setups when we don't care about where the server will write its records.

I'll add a shutdown hook to clean up the storage directory if it's being created here.